libs/execution_context/ebpf_program.c

```diff
@@ -200,132 +206,6 @@ _ebpf_program_program_info_provider_changed(
-                program->global_helper_function_count,
-                &total_helper_count);
-            if (return_value != EBPF_SUCCESS) {
-                goto Exit;
-            }
-
-            total_helper_function_addresses =
-                (ebpf_helper_function_addresses_t*)ebpf_allocate(sizeof(ebpf_helper_function_addresses_t));
-            if (total_helper_function_addresses == NULL) {
-                return_value = EBPF_NO_MEMORY;
-                goto Exit;
-            }
-            total_helper_function_addresses->helper_function_count = (uint32_t)total_helper_count;
-            total_helper_function_addresses->helper_function_address =
-                (uint64_t*)ebpf_allocate(sizeof(uint64_t) * total_helper_count);
-            if (total_helper_function_addresses->helper_function_address == NULL) {
-                return_value = EBPF_NO_MEMORY;
-                goto Exit;
-            }
-
-            if (!program->trampoline_table) {
-                // Program info provider is being loaded for the first time. Allocate trampoline table.
-                return_value = ebpf_allocate_trampoline_table(total_helper_count, &program->trampoline_table);
-                if (return_value != EBPF_SUCCESS)
-                    goto Exit;
-            }
-
-            __analysis_assume(total_helper_count > 0);
-            total_helper_function_ids = (uint32_t*)ebpf_allocate(sizeof(uint32_t) * total_helper_count);
-            if (total_helper_function_ids == NULL) {
-                return_value = EBPF_NO_MEMORY;
-                goto Exit;
-            }
-
-            if (helper_function_addresses != NULL) {
-                helper_prototypes = program_info->program_type_specific_helper_prototype;
-                if (helper_prototypes == NULL) {
-                    EBPF_LOG_MESSAGE_GUID(
-                        EBPF_TRACELOG_LEVEL_ERROR,
-                        EBPF_TRACELOG_KEYWORD_PROGRAM,
-                        "program_info->program_type_specific_helper_prototype can not be NULL",
-                        program->parameters.program_type);
-                    return_value = EBPF_INVALID_ARGUMENT;
-                    goto Exit;
-                }
-
-#pragma warning(push)
-#pragma warning(disable : 6386) // Buffer overrun while writing to 'total_helper_function_ids'.
-                for (uint32_t index = 0; index < program->program_type_specific_helper_function_count; index++) {
-                    total_helper_function_ids[index] = helper_prototypes[index].helper_id;
-                    total_helper_function_addresses->helper_function_address[index] =
-                        helper_function_addresses->helper_function_address[index];
-                }
-            }
-#pragma warning(pop)
-
-            if (global_helper_function_addresses != NULL) {
-                helper_prototypes = program_info->global_helper_prototype;
-                if (helper_prototypes == NULL) {
```

Comment:
Same for this check. I think we should keep these for sanity checks and catch any bugs in the extension?

Reply:
https://github.com/microsoft/ebpf-for-windows/issues/1910